Set root time to 1 if all times < 1

hyanwong opened the pull request:
It's confusing that, with the "time" of ancestors in 0..1, the root ancestor is placed at just under "time" 2, and the ultimate ancestor at ~ time 3. Here's a quick hack to make the root ancestor at time 1 if possible (normally we shouldn't have any ancestors at time 1, because that would imply the variant was fixed), and the ultimate ancestor at a fraction more than 1. This should help when plotting ancestor times, which can look a bit wacky otherwise.

jeromekelleher commented on tsinfer/inference.py:
```diff
@@ -947,8 +947,12 @@ def run(self):
+            if root_time < 1:
+                root_time = 1  # The normal case, when time==freq
+            else:
+                root_time += 1  # Could happen with user-specified times
```
I don't understand this comment - what could happen?

hyanwong replied:
We could have root_time >=1 (because the root time could be any number)

jeromekelleher commented on tsinfer/inference.py:
```diff
@@ -947,8 +947,12 @@ def run(self):
+                root_time = 1  # The normal case, when time==freq
+            else:
+                root_time += 1  # Could happen with user-specified times
+            ultimate_ancestor_time = root_time + 1 / self.num_ancestors
```
Why the new 1/N fraction?

hyanwong replied:
What do you suggest using instead? I could use root_time+1, as before, which would mean in the usual case most nodes would be at times `[0..1)`, apart from the root (at time 1), and the ultimate ancestor (at time 2). This creates weird plots with all node times 0..1 apart from one at time 2.

If we use `np.nextafter`, the augment_ancestors functionality fails, as it adds 1 to all the times. So `1 / self.num_ancestors` seemed like a reasonable epsilon.

jeromekelleher reviewed (commented):
Seems OK to me. Needs a bit more clarity in the comments and a couple of simple tests to check the behaviour.

jeromekelleher reviewed (commented):
LGTM, just need to udpate the CHANGELOG with this detail (or we'll forget it)

hyanwong commented:
> LGTM, just need to udpate the CHANGELOG with this detail (or we'll forget it)

Great point. Updated (and I added the change for map_mutations too)

jeromekelleher commented on CHANGELOG.rst:
```diff
@@ -1,3 +1,21 @@
+
+**Bugfixes**:
+
+- Placement of mutations at non-inference sites was not always truely parsimonious,
```
truly

jeromekelleher replied:
Maybe "fully parsimonious" would be a better description. Also this won't make sense to an outsider: "Mutations at non-inference sites are now guaranteed to be fully parsimonious. Previous versions required a mutation above the root when the input ancestral state disagreed with the ancestral state produced by the parsimony algorithm."

hyanwong replied:
Yep, sounds good. I was probably dozy when I wrote it.

jeromekelleher commented on CHANGELOG.rst:
```diff
@@ -1,3 +1,21 @@
+
+**New Features**:
+
+- Oldest nodes in a standard inferred tree sequence are no longer set to frequencies ~2
```
This isn't a feature, it's a breaking change - people might be depending on the old behaviour.

jeromekelleher reviewed (commented):
Needs a bit of a rephrase - the CHANGELOG is for users who don't closely follow development to read and be informed of the important changes.
